core/src/main/scala/akka/persistence/cassandra/Extractors.scala

```diff
@@ -0,0 +1,193 @@
+ */
+
+package akka.persistence.cassandra
+
+import com.datastax.oss.driver.api.core.cql.Row
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import akka.persistence.PersistentRepr
+import akka.persistence.cassandra.journal._
+import akka.persistence.cassandra.journal.CassandraJournal._
+import akka.annotation.InternalApi
+import java.{ util => ju }
+import akka.util.OptionVal
+import akka.serialization.Serialization
+import scala.collection.JavaConverters._
+import java.nio.ByteBuffer
+import com.datastax.oss.protocol.internal.util.Bytes
+
+/**
+ * An Extractor takes reads a row from the messages table. There are different extractors
+ * for creating PersistentRepr, TaggedPersistentRepr, and RawEvent.
+ *
+ * INTERNAL API
+ */
+@InternalApi private[akka] object Extractors {
+
+  /**
+   * A [[PersistentRepr]] along with its tags and offset.
+   */
+  case class TaggedPersistentRepr(pr: PersistentRepr, tags: Set[String], offset: ju.UUID) {
+    def sequenceNr: Long = pr.sequenceNr
+  }
+
+  case class OptionalTagged(sequenceNr: Long, tagged: OptionVal[TaggedPersistentRepr])
+
+  /**
+   * A RawEvent is an event from the messages table in its serialized form
+   * Used for reading and writing between the messages table and the tag_views
+   * table without serializing/de-serializing
+   */
+  case class RawEvent(sequenceNr: Long, serialized: Serialized)
+
+  abstract class Extractor[T] {
+    def extract(row: Row, async: Boolean)(implicit ec: ExecutionContext): Future[T]
+  }
+
+  final case class SeqNrValue(sequenceNr: Long)
+
+  final case class PersistentReprValue(persistentRepr: PersistentRepr) {
+    def sequenceNr: Long = persistentRepr.sequenceNr
+  }
+
+  /**
+   * Extractor that does not de-serialize the event.
+   * It does not support versions older than 0.60 that serialized
+   * the PersistentRepr into the message column
+   *
+   * @param bucketSize for calculating which bucket each event should
+   */
+  def rawEvent(bucketSize: BucketSize): Extractor[RawEvent] = {
+    new Extractor[RawEvent] {
+
+      // Could  make this an extension? Too global?
+      val columnDefinitionCache = new ColumnDefinitionCache
+
+      override def extract(row: Row, async: Boolean)(implicit ec: ExecutionContext): Future[RawEvent] = {
+        val tags = extractTags(row, columnDefinitionCache)
+        val timeUuid = row.getUuid("timestamp")
+        val sequenceNr = row.getLong("sequence_nr")
+        val meta = if (columnDefinitionCache.hasMetaColumns(row)) {
+          val m = row.getByteBuffer("meta")
+          Option(m).map(SerializedMeta(_, row.getString("meta_ser_manifest"), row.getInt("meta_ser_id")))
+        } else {
+          None
+        }
+
+        Future.successful(
+          RawEvent(
+            sequenceNr,
+            Serialized(
+              row.getString("persistence_id"),
+              row.getLong("sequence_nr"),
+              row.getByteBuffer("event"),
```

Comment:
in other places we also check for the old "message" column. See usages of `eventDeserializer.columnDefinitionCache.hasMessageColumn`

Reply:
I guess we should support data from that far back,  means we it also needs to know about serialization